### DISCUSSION TITLE
CentOS 8 Life cycle

### DISCUSSION THREAD

**Discussion by stx-hseow** [Seagate/cortx#896] [Q&A] [answered]
It's been quite a while since the announcement of CentOS 8 EOL ([Table](https://wiki.centos.org/About/Product)) scheduled by December 2021.
Will Cortx stick to CentOS 7 (according to [LocalVMSetup](https://github.com/Seagate/cortx/blob/main/doc/LocalVMSetup.md)) for now while looking at other Linux distro or would go for CentOS Stream?

Consideration for [Rocky Linux](https://github.com/rocky-linux/rocky)?

**Comment by mukul-seagate11** [comment]
@shailesh-vaidya, can you address Harrison query as I see CentOS 8 in the roadmap but do we have any plan defined? I hope this might be a ARC call ?

**Comment by justinzw** [comment]
Thanks @harrison-seow-seagate I agree we should probably support the latest CentOS but what are the benefits of supporting Rocky Linux over CentOS.

Thanks for suggesting this!

**Comment by stx-hseow** [comment]
Hi @justinzw, the latest CentOS 8 will have no updates/maintenance by Dec 2021 (unless further migrate to CentOS Steam).
CentOS is famous for being a reliable Linux distribution (downstream of RHEL), but CentOS Stream is a development (upstream) branch of RHEL.

I would say it's generally a bad idea to use development branch for production use, also advised by Red Hat, to go for the paid RHEL if stability is needed.

Rocky Linux is intended to be 100% bug-to-bug compatible with Enterprise Linux, effectively replacing the gap left by CentOS 8 after its EOL. (_Though I can't say much about it in the current state as it's still new, but it's initiated by the founder of CentOS_)

I'm sure there are other people with better insights, but I would personally think CentOS Stream is probably not the best option.

[a comment by an automated account is not shown]

**Comment by stx-hseow** [comment]
I'm not sure if this is categorized as an issue, perhaps this can be moved to discussion?

_However, I think it would be an issue if migration to CentOS 8 has started._

**Comment by mukul-seagate11** [comment]
> I'm not sure if this is categorized as an issue, perhaps this can be moved to discussion?
> 
> _However, I think it would be an issue if migration to CentOS 8 has started._

It will be a discussion as the migration to CentOS 8 yet not started

**Comment by justinzw** [comment, marked as answer]
@harrison-seow-seagate thanks again for bringing this up. 

Currently CentOS 8 End of Support is expected in Dec 2021 (https://wiki.centos.org/About/Product).

We're staying with CentOS 7.9 now (supported till Jun'24), and are actively considering alternatives and would love to hear any thoughts you and others have.

**Reply by mukul-seagate11** [reply]
yes as per CentOS 7.9 EOL, https://endoflife.software/operating-systems/linux/centos

**Reply by andriytk** [reply]
Maybe we can stick to RHEL-based Oracle Linux which still seems to provide [10-years](https://www.oracle.com/a/ocom/docs/elsp-lifetime-069338.pdf) of major release version support for free, the same way as CentOS used to do?

> Q. Wait, doesn't Oracle Linux cost money?
> A. Oracle Linux support costs money. If you just want the software, it's 100% free. And it's all in our yum repo at yum.oracle.com. Major releases, errata, the whole shebang. Free source code, free binaries, free updates, freely redistributable, free for production use. Yes, we know that this is Oracle, but it's actually free. Seriously.

https://linux.oracle.com/switch/centos/
https://blogs.oracle.com/linux/reasons-for-switching-centos-to-oracle-linux

Another option maybe can be Ubuntu Server LTS, but it has only 5 years of major release version free support. Not sure if it's enough. [RHEL-based distributions](https://en.wikipedia.org/wiki/Red_Hat_Enterprise_Linux_derivatives#Notable_Red_Hat_Enterprise_Linux_derivatives) seem to be the only ones out there to provide 10+ years of major releases support, simply by following the RHEL's release support policy.

BTW, Ubuntu build PR for Motr is upcoming - https://github.com/Seagate/cortx-motr/pull/540.
CentOS-8 just has been landed - https://github.com/Seagate/cortx-motr/pull/523.

**Reply by mukul-seagate11** [reply]
@andriytk, thanks for the suggestions but this is still in discussion phase where we have to adopt a better option from long term perspective.

**Reply by andriytk** [reply]
Better in what terms? @justinzw asked for thoughts/suggestions, people should know what to suggest.

I used the following criteria:

1. Long 10+ years lifecycle of major release versions.
2. Free and paid subscriptions (in case support is needed).
3. Arm64 builds available (along with x86_64).

**Reply by mukul-seagate11** [reply]
@andriytk, as I said earlier that this is in discussion where Architecture Board will take the final call as its not a easy tasks, it involves lot of changes from design/migration/dependencies etc.

**Reply by andriytk** [reply]
Which can be another, 4th criteria - to avoid a lot of changes during the migration.
